Content.Server/_DV/Silicons/StationAiShopSystem.cs

```diff
@@ -0,0 +1,78 @@
+using Content.Server.Fluids.EntitySystems;
+using Content.Server.Light.EntitySystems;
+using Content.Server.Spreader;
+using Content.Server.Store.Systems;
+using Content.Shared._DV.Silicons;
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Coordinates.Helpers;
+using Content.Shared.Maps;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Map;
+
+namespace Content.Server._DV.Silicons;
+
+public sealed class StationAiShopSystem : SharedStationAiShopSystem
+{
+    [Dependency] private readonly StoreSystem _store = default!;
+    [Dependency] private readonly IMapManager _map = default!;
+    [Dependency] private readonly SmokeSystem _smoke = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly SpreaderSystem _spreader = default!;
+    [Dependency] private readonly PoweredLightSystem _poweredLight = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<StationAiShopComponent, StationAiShopActionEvent>(OnShopAction);
+
+        SubscribeLocalEvent<StationAiShopComponent, StationAiLightSynthesizerActionEvent>(OnLightSynthesizer);
+        SubscribeLocalEvent<StationAiShopComponent, StationAiSmokeActionEvent>(OnEmergencySealant);
+    }
+
+    private void OnShopAction(Entity<StationAiShopComponent> ent, ref StationAiShopActionEvent args)
+    {
+        _store.ToggleUi(args.Performer, ent);
+    }
+
+    private void OnLightSynthesizer(Entity<StationAiShopComponent> ent, ref StationAiLightSynthesizerActionEvent args)
+    {
+        if (_poweredLight.EjectBulb(args.Target) is { } oldBulb)
+            Del(oldBulb);
+
+        var bulb = Spawn(args.BulbPrototype);
+        if (!_poweredLight.InsertBulb(args.Target, bulb))
+        {
+            Del(bulb);
+            return;
+        }
+
+        args.Handled = true;
+    }
+
+    private void OnEmergencySealant(Entity<StationAiShopComponent> ent, ref StationAiSmokeActionEvent args)
+    {
+        var mapCoords = _transform.ToMapCoordinates(args.Target);
+        if (!_map.TryFindGridAt(mapCoords, out _, out var grid) ||
+            !grid.TryGetTileRef(args.Target, out var tileRef) ||
+            tileRef.Tile.IsEmpty)
+        {
+            return;
+        }
+
+        if (_spreader.RequiresFloorToSpread(args.SmokePrototype.ToString()) && tileRef.Tile.IsSpace())
+            return;
+
+        var coords = grid.MapToGrid(mapCoords);
+        var uid = Spawn(args.SmokePrototype, coords.SnapToGrid());
+        if (!TryComp<SmokeComponent>(uid, out var smoke))
+        {
+            Log.Error($"Smoke prototype {args.SmokePrototype} was missing SmokeComponent");
+            Del(uid);
+            return;
+        }
```

Comment:
use `EntProtoId<SmokeComponent>` instead of reinventing the wheel, dont need to pass component its resolved anyway